src/features/wallet/root/useActivityItem.tsx

```diff
@@ -0,0 +1,299 @@
+            ? t('transactions.sent')
+            : t('transactions.received')
+        case 'assert_location_v1':
+          return t('transactions.location')
+        case 'transfer_hotspot_v1':
+          return isSelling(item)
+            ? t('transactions.transferSell')
+            : t('transactions.transferBuy')
+        case 'rewards_v1':
+          return t('transactions.mining')
+        case 'token_burn_v1':
+          return t('transactions.burnHNT')
+      }
+    },
+    [isSending, isSelling, t],
+  )
+
+  const detailIcon = useCallback(
+    (item: AnyTransaction | PendingTransaction) => {
+      switch (item.type as TxnType) {
+        case 'transfer_hotspot_v1':
+          return <HotspotTransfer height={20} width={50} />
+        case 'payment_v1':
+        case 'payment_v2':
+          return isSending(item) ? (
+            <SentHnt width={35} height={24} />
+          ) : (
+            <ReceivedHnt width={35} height={24} />
+          )
+        case 'assert_location_v1':
+          return <Location width={20} height={23} />
+        case 'rewards_v1':
+          return <Rewards width={26} height={26} />
+        case 'token_burn_v1':
+          return <Burn width={23} height={28} />
+        case 'add_gateway_v1':
+        default:
+          return <HotspotAdded width={20} height={20} />
+      }
+    },
+    [isSending],
+  )
+
+  const listIcon = useCallback(
+    (item: AnyTransaction | PendingTransaction) => {
+      switch (item.type as TxnType) {
+        case 'payment_v1':
+        case 'payment_v2':
+          return isSending(item) ? (
+            <SentHnt width={32} height={18} />
+          ) : (
+            <ReceivedHnt width={32} height={18} />
+          )
+        case 'assert_location_v1':
+          return <Location width={20} height={23} />
+        case 'rewards_v1':
+          return <Rewards width={26} height={26} />
+        case 'token_burn_v1':
+          return <Burn width={23} height={28} />
+        case 'transfer_hotspot_v1':
+        case 'add_gateway_v1':
+        default:
+          return <HotspotAdded width={20} height={20} />
+      }
+    },
+    [isSending],
+  )
+
+  const isFee = useCallback(
+    (item: AnyTransaction | PendingTransaction) => {
+      if (item instanceof PaymentV1 || item instanceof PaymentV2) {
+        return isSending(item)
+      }
+
+      if (item instanceof RewardsV1) {
+        return false
+      }
+
+      if (item instanceof TransferHotspotV1) {
+        return (item as TransferHotspotV1).seller === address
+      }
+
+      return true
+    },
+    [isSending, address],
+  )
+
+  const formatAmount = (
+    fee: boolean,
+    amount?: Balance<DataCredits | NetworkTokens> | number,
+  ) => {
+    if (!amount) return ''
+
+    if (typeof amount === 'number') {
+      if (amount === 0) return '0'
+      return `${fee ? '-' : '+'}${amount.toLocaleString()}`
+    }
+
+    if (amount?.floatBalance === 0) return amount.toString()
```

Comment:
Seeing some long numbers here. We can make it `amount.toString(8)` to make sure its not showing too many decimals